examples/linear_model/plot_tweedie_regression_insurance_claims.py

```diff
@@ -0,0 +1,593 @@
+"""
+======================================
+Tweedie regression on insurance claims
+======================================
+
+This example illustrates the use of Poisson, Gamma and Tweedie regression on
+the French Motor Third-Party Liability Claims dataset, and is inspired by an R
+tutorial [1].
+
+Insurance claims data consist of the number of claims and the total claim
+amount. Often, the final goal is to predict the expected value, i.e. the mean,
+of the total claim amount per exposure unit also referred to as the pure
+premium.
+
+There are several possibilities to do that, two of which are:
+
+1. Model the number of claims with a Poisson distribution, the average claim
+   amount per claim, also known as severity, as a Gamma distribution and
+   multiply the predictions of both in order to get the total claim amount.
+2. Model total claim amount directly, typically with a Tweedie distribution of
+   Tweedie power :math:`p \\in (1, 2)`.
+
+In this example we will illustrate both approaches. We start by defining a few
+helper functions for loading the data and visualizing results.
+
+.. [1]  A. Noll, R. Salzmann and M.V. Wuthrich, Case Study: French Motor
+    Third-Party Liability Claims (November 8, 2018). `doi:10.2139/ssrn.3164764
+    <http://dx.doi.org/10.2139/ssrn.3164764>`_
+
+"""
+print(__doc__)
+
+# Authors: Christian Lorentzen <lorentzen.ch@gmail.com>
+#          Roman Yurchak <rth.yurchak@gmail.com>
+#          Olivier Grisel <olivier.grisel@ensta.org>
+# License: BSD 3 clause
+from functools import partial
+
+import numpy as np
+import matplotlib.pyplot as plt
+import pandas as pd
+
+from sklearn.datasets import fetch_openml
+from sklearn.compose import ColumnTransformer
```

Comment:
No strong opinion on this suggestion.

Reply:
> maybe we could delay the import in the place that they are used for the first time?

sphinx will autolink to class definition in places where they are used in any case. I think it's good for the examples to be realistic and realistically imports are typically at the top. Otherwise beginner users might start doing  imports all over their files.